src/Checkbox.js

```diff
@@ -1,69 +1,55 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 
-class Checkbox extends Component {
-  constructor(props) {
-    super(props);
-  }
+const Checkbox = ({
+  id,
+  className,
+  indeterminate,
+  filledIn,
+  label,
+  onChange,
+  ...props
+}) => {
+  const [checked, setChecked] = useState(props.checked);
+  const _input = useRef(null);
 
-  componentDidMount() {
-    const { indeterminate } = this.props;
-
-    if (this._input && indeterminate) {
-      this._input.indeterminate = indeterminate;
-      this._input.checked = false;
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.indeterminate !== this.props.indeterminate) {
-      this._input.indeterminate = this.props.indeterminate;
+  useEffect(() => {
+    if (_input.current) {
+      _input.current.indeterminate = indeterminate;
+      _input.current.checked = false;
+      setChecked(false);
```

Comment:
@alextrastero We now correctly set the `state` to `false` when changing `undeterminate`